datafusion/physical-expr/src/expressions/column.rs

```diff
@@ -33,32 +33,67 @@ use datafusion_expr::ColumnarValue;
 use crate::physical_expr::{down_cast_any_ref, PhysicalExpr};
 
 /// Represents the column at a given index in a RecordBatch
+///
+/// This is a physical expression that represents a column at a given index in a
```

Comment:
The point that physical exprs are always in terms of indexes wasn't as clear as I think it could be so I tried to improve that